include/Bezier/bezier.h

```diff
@@ -280,15 +314,15 @@ class Curve
   Eigen::MatrixX2d control_points_;
 
   // private caching
-  std::shared_ptr<const Curve> cached_derivative_; /*! If generated, stores derivative for later use */
-  std::unique_ptr<PointVector> cached_roots_;      /*! If generated, stores roots for later use */
-  std::tuple<double, double, std::size_t> cached_roots_params_{0, 0, 0}; /*! epsilon and max_iter of cached roots */
-  std::unique_ptr<BoundingBox>
-      cached_bounding_box_tight_; /*! If generated, stores bounding box (use_roots = true) for later use */
-  std::unique_ptr<BoundingBox>
-      cached_bounding_box_relaxed_; /*! If generated, stores bounding box (use_roots = false) for later use */
-  std::unique_ptr<PointVector> cached_polyline_;            /*! If generated, stores polyline for later use */
-  std::tuple<double, double> cached_polyline_params_{0, 0}; /*! Smootheness and precision of cached polyline */
+  std::shared_ptr<const Curve> cached_derivative_;         /*! If generated, stores derivative for later use */
+  std::unique_ptr<ParameterVector> cached_roots_;          /*! If generated, stores roots for later use */
```

Comment:
Is there a reason why this is unique_ptr?

Reply:
pointer because cache might not exist, unique_ptr because there is only 1 owner (class instance), and additionally, for less hassle with resetting/changing the cache

Reply:
looks like my hunch was right, found a question about it on the forum: https://forum.kde.org/viewtopic.php?f=74&t=97255